### PR TITLE
Disabling ConvOclBwdWrW2  for asymmetric padding 

### PR DESCRIPTION
**Pull request by shurale-nkn**
Added condition in IsApplicable.

Fix for issue #142

### CODE REVIEW

**Comment by daniellowell** [comment]
We need to purge the perf db of entries with this solver with any input that have asymmetric padding.

**Comment by daniellowell** [comment]
@JehandadKhan @alexandraBara For awareness.

**Comment by atamazov** [comment]
> We need to purge the perf db of entries with this solver with any input that have asymmetric padding.

There is no harm in this. Do you mean find-db?

File: src/include/miopen/conv/problem_description.hpp [hunk 1]
```diff
@@ -235,6 +235,15 @@ struct ProblemDescription
     std::size_t GetBackwardPadW() const { return GetWeightsWidth() - GetPadW() - 1; }
     std::size_t GetBackwardPadH() const { return GetWeightsHeight() - GetPadW() - 1; }
 
+    bool IsAsymmetricPadH() const
+    {
+        return conv.paddingMode == miopenPaddingSame && (GetWeightsHeight() % 2) == 0;
```
Comment (atamazov) [inline src/include/miopen/conv/problem_description.hpp:240]: Please check code around https://github.com/ROCmSoftwarePlatform/MIOpen/blob/cb1ef1f2a89ebbc4e556cce51234456d9dc9ea03/src/convolution.cpp#L242-L248

It seems like in certain cases we can get asymmetric padding even in miopenPaddingDefault mode.

**Comment by atamazov** [comment]
IIRC none of OCL kernels support asymmetric padding. And some old asm kernels too. @Kirpich30000 do you remember?

File: src/include/miopen/conv/problem_description.hpp (hunk 1 above)
Comment (shurale-nkn) [reply src/include/miopen/conv/problem_description.hpp:240]: 2 * GetConvPads()[i] - is symmetric padding;
1 - case where filter bigger than input image;

**Comment by Kirpich30000** [comment]
No, I don't remember details about OCL kernels

**Comment by shurale-nkn** [comment]
@atamazov OCL do not support even filters or were disabled earlier. 
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/cb1ef1f2a89ebbc4e556cce51234456d9dc9ea03/src/solver/conv_ocl_dir2D_bwdWrW_53.cpp#L103-L110

**Comment by atamazov** [comment]
@Kirpich30000  
> No, I don't remember details about OCL kernels

What about asm kernels?

**Comment by atamazov** [comment]
Further TODOs:
- Disable asymmetric padding for all OCL kernels
- Disable asymmetric padding for asm kernels that do not support it (pending @Kirpich30000 )

File: src/include/miopen/conv/problem_description.hpp [hunk 2]
```diff
@@ -235,6 +235,15 @@ struct ProblemDescription
...
+    }
+    bool IsAsymmetricPadW() const
```
Comment (atamazov) [inline src/include/miopen/conv/problem_description.hpp:242]: Do these methods provide valid answer for all directions (Fwd/Bwd/WrW)?

**Comment by Kirpich30000** [comment]
Single and multi pass Winograd works with any padding. Other kernels were designed for symmetrical padding.

**Comment by daniellowell** [comment]
@alexandraBara @JehandadKhan @cderb For awareness.

File: src/include/miopen/conv/problem_description.hpp (hunk 2 above)
Comment (atamazov) [reply src/include/miopen/conv/problem_description.hpp:242]: @shurale-nkn @Kirpich30000 please answer

**Pull request merged**